framework/src/base/Assembly.C

```diff
@@ -1396,42 +1431,45 @@ Assembly::cacheResidualNodes(DenseVector<Number> & res, std::vector<dof_id_type>
   // Add the residual value and dof_index to cached_residual_values and cached_residual_rows
   // respectively.
   // This is used by NodalConstraint.C to cache the residual calculated for master and slave node.
-  Moose::KernelType type = Moose::KT_NONTIME;
   for (unsigned int i = 0; i < dof_index.size(); ++i)
   {
-    _cached_residual_values[type].push_back(res(i));
-    _cached_residual_rows[type].push_back(dof_index[i]);
+    _cached_residual_values[0].push_back(res(i));
```

Comment:
We can't have `0` here like this.  If this should be `nontime` then we need to pull out the `nontime` tag.  Even better: we should pass the tag into this function.